### ISSUE TITLE
Swallowing errors in electron

### ISSUE THREAD

**Issue by Mosho1** [systemjs/systemjs#1463]
I'm not sure this is specific to electron. I'm building an app with it and systemjs, and I seem to have tried to import `typescript.js` as AMD, which it isn't, and the error I received was this:

```
Error loading file:///C:/repos/electron-quick-start/node_modules/typescript/lib/typescript.js as "typescript" from file:///C:/repos/electron-quick-start/node_modules/plugin-typescript/lib/plugin.js
```

while the first argument to `addToError(err, msg)` was an error with message:

```
"AMD module file:///C:/repos/electron-quick-start/node_modules/typescript/lib/typescript.js did not define"
```

It seems that this condition in `addToError` evaluates to false and causes the errors not to be passed on:

```
if (typeof $__curScript == 'undefined' || stack[i].indexOf($__curScript.src) == -1)
```

I'm using 0.19.39 on windows 10, on node 6.5.0 (and electron 1.4.1)

**Comment by guybedford** [comment]
Thanks for posting. Can you confirm what the value of `$__curScript` is in this scenario?

**Comment by guybedford** [comment]
I mean `$__curScript.src` - it seems to be filtering it out as the loader path for some reason, which may be an incorrect detection of `$__curScript`.

**Comment by Mosho1** [comment]
It's `$__curScript.src` is `""` and `$__curScript` points to the script here: (this is the view html file for the electron app)

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Hello World!</title>
    <link rel="stylesheet" href="style.css"/>
  </head>
  <body>
    <div id="app">
    </div>
  </body>

  <script>
    // You can also require other files to run in this process
    require('./renderer.js')
  </script>
</html>
```

**Comment by guybedford** [comment]
I've added a possible fix in https://github.com/systemjs/systemjs/commit/232f994b744d45b91342b38179453e3163eee2ab. Let me know if that works for you.

**Comment by Mosho1** [comment]
It gets re-assigned before reaching the point of the error here (line 1567):

```
if (isBrowser && typeof document != 'undefined' && document.getElementsByTagName) {
    var scripts = document.getElementsByTagName('script');
    $__curScript = scripts[scripts.length - 1];

    if (!(window.chrome && window.chrome.extension || navigator.userAgent.match(/^Node\.js/)))
      supportsScriptExec = true;
  }
```

**Comment by guybedford** [comment]
Thanks, I've updated that codepath which I believe was made redundant by changes in the wrapper previously.

**Comment by Mosho1** [comment]
That did it! thanks.

**Comment by guybedford** [comment]
Released in 0.19.40.
